packages/amplify_storage_s3/example/ios/.symlinks/plugins/amplify_core/ios/Classes/Support/AmplifyFlutterValidationException.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+import AmplifyPlugins
+import AWSPluginsCore
+
+class AmplifyFlutterValidationException: AmplifyError {
+
+    var errorDescription: ErrorDescription
+
+    var recoverySuggestion: RecoverySuggestion
+
+    var underlyingError: Error?
+
+    required init(errorDescription: ErrorDescription, recoverySuggestion: RecoverySuggestion, error: Error = AWSCognitoAuthError.invalidParameter) {
+        self.errorDescription = errorDescription
+        self.recoverySuggestion = recoverySuggestion
+    }
```

Comment:
`underlyingError` doesn't seem to be getting updated here.

Reply:
We aren't using underlyingError in this case.

Reply:
Sorry I'm missing the point here. if underlyingError can never be initialized why do we need it? Is this a requirement when you are extending a class or adhering to a Protocol?

Reply:
This is a requirement... but I think I just realized a seperate issue that I can resolve here.